pkg/plugins/generic/generic_plugin.go

```diff
@@ -26,6 +26,7 @@ type GenericPlugin struct {
 	LastState            *sriovnetworkv1.SriovNetworkNodeState
 	LoadVfioDriver       uint
 	LoadVirtioVdpaDriver uint
+	DesiredKernelParams  map[string]uint
```

Comment:
in general I don't think we need this if we want to be sure it's loaded lets just check the proc cmdline file.

Reply:
We need to record what kernel parameters we want to load somewhere. It would be nice for this to happen anywhere in the generic plugin. This is why I think it is still needed.

Reply:
> @zeeke Could you elaborate more on why adding a state to GeneralPlugin will make debugging sessions harder?

When looking at logs, we will not be able to fully understand the state of the operator, as it will have internal variables.

> Could you explain by adding the map and functions in its own private package, what exactly that helps to do?

It improves the design, as part of logic encapsulation. Making that public means they are meant to be called by other packages, while I suppose it's a bad idea to call `SetAllDesiredKernelParams()` and `AddToDesiredKernelParams()` from other components, isn't it?

Reply:
@zeeke I see what you mean by adding state variables is not very good. I would like to compare `rpm-ostree kargs` vs. `/proc/cmdline`. BUT even when rebooting there is no grantee that `rpm-ostree kargs = /proc/cmdline`. I believe this is what @SchSeba was also thinking. But in some environments:
`--import-proc-cmdline             Instead of modifying old kernel arguments, we modify args from current /proc/cmdline (the booted deployment)`
is not used with `rpm-ostree kargs` (e.g. OCP).

I made the functions private to the the GenericPlugin such that they can't be called externally. Thanks for catching this!